src/main/scala/scala/compat/java8/PrimitiveIteratorConversions.scala

```diff
@@ -0,0 +1,108 @@
+package scala.compat.java8
+
+import language.implicitConversions
+import java.util.{ Iterator => JIterator, PrimitiveIterator }
+
+/** This class enables conversion from `scala.Iterator` to the set of
+  * `java.util.PrimitiveIterator` classes.
+  *
+  * Scala's `Iterator` is generic, as is its `java.util` counterpart.  However,
+  * `java.util.PrimitiveIterator` offers three manually-specialized variants of
+  * `Iterator`: `OfDouble`, `OfInt`, and `OfLong`.  This class provides
+  * `.asPrimitive` extension methods for Scala and Java iterators to present
+  * the generic versions as the specialized version.
+  *
+  * Example usage:
+  *
+  * {{{
+  * import scala.compat.java8.PrimitiveIteratorConverters._
+  * val it = Iterator(1.0, 2.0, math.Pi)
+  * val jpid = it.asPrimitive   // PrimitiveIterator.OfDouble
+  * }}}
+  */
+object PrimitiveIteratorConverters {
```

Comment:
Is this intentionally in a different package than the wrapper classes themselves?